sdk/eventhub/Azure.Messaging.EventHubs/MIGRATIONGUIDE.md

```diff
@@ -0,0 +1,357 @@
+  - [Migrating code from `EventHubClient` to `EventHubProducerAsyncClient` for publishing events with partition key](#migrating-code-from-eventhubclient-to-eventhubproducerclient-for-publishing-events-with-partition-key)
+  - [Migrating code from `PartitionReceiver` to `EventHubConsumerAsyncClient` for reading events in batches](#migrating-code-from-partitionreceiver-to-eventhubconsumerclient-for-reading-events-in-batches)
+  - [Migrating code from `EventProcessorHost` to `EventProcessorClient` for reading events](#migrating-code-from-eventprocessorhost-to-eventprocessorclient-for-reading-events)
+- [Additional samples](#additional-samples)
+
+## General changes
+
+In the interest of simplifying the API surface we've made two distinct
+clients, rather than having a single `EventHubClient`:
+* [EventHubProducerClient](https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.eventhubs.eventhubproducerclient?view=azure-dotnet-preview)
+  for publishing messages.
+* [EventHubConsumerClient](https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.eventhubs.eventhubconsumerclient?view=azure-dotnet-preview) 
+  for reading messages.
+
+The producer and consumer clients operate in the context of a specific event hub and offer operations for all partitions. Unlike the v4, the clients are not bound to a specific partition, but the methods on them have overloads to handle specific partitions if needed.
+
+The [EventHubConsumerClient](https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.eventhubs.eventhubconsumerclient?view=azure-dotnet-preview) supports reading events from a single partition and also offers an easy way to familiarize yourself with Event Hubs by reading from all partitions without the rigor and complexity that you would need in a production application. For reading events from all partitions in a production scenario, we strongly recommend using the [EventProcessorClient](https://azuresdkdocs.blob.core.windows.net/$web/dotnet/Azure.Messaging.EventHubs.Processor/5.0.0-preview.6/api/Azure.Messaging.EventHubs/Azure.Messaging.EventHubs.EventProcessorClient.html).
+
+We've introduced a new library [Azure.Messaging.EventHubs.Processor](https://www.nuget.org/packages/Azure.Messaging.EventHubs.Processor/) which replaces the older [Microsoft.Azure.EventHubs.Processor](https://www.nuget.org/packages/Microsoft.Azure.EventHubs.Processor/). Here we have the [EventProcessorClient](https://azuresdkdocs.blob.core.windows.net/$web/dotnet/Azure.Messaging.EventHubs.Processor/5.0.0-preview.6/api/Azure.Messaging.EventHubs/Azure.Messaging.EventHubs.EventProcessorClient.html) which is responsible for consuming events for the configured Event Hub and consumer group across all partitions. It also supports checkpointing and load balancing. Currently, only Azure Storage Blobs is supported for checkpointing.
```

Comment:
(extreme) nit:  We may want to leave with the statement about the processor client as a natural segue from the previous paragraph and then branch into "The EventProcessorClient can be found in the new Azure.Messaging.EventHubs.Processor package that replaces. ..."